ReactWindows/ReactNative.Shared/Views/Text/ReactSpanViewManager.cs

```diff
@@ -110,6 +112,50 @@ public void AddView(DependencyObject parent, DependencyObject child, int index)
 
 #if WINDOWS_UWP
             span.Inlines.Insert(index, inlineChild);
+
+            // Initialize ImportantForAccessibility for the child.
+            // Post on UI thread because the method requires that parent/child
+            // relationship is established but XAML does not set the relationship
+            // until the next loop after the child is added.
+            DispatcherHelpers.RunOnDispatcher(() =>
+            {
+                // Find the UIElement that owns the Span. If the Span has not been added
+                // to an UIElement yet, accessing ElementStart will throw NotSupportedException.
+                DependencyObject spanParent;
+                try
+                {
+                    spanParent = span.ElementStart?.Parent;
+                }
+                catch (NotSupportedException)
+                {
+                    return;
+                }
+
+                while (spanParent is TextElement textElementParent)
+                {
+                    try
+                    {
+                        spanParent = textElementParent.ElementStart?.Parent;
+                    }
+                    catch (NotSupportedException)
+                    {
+                        return;
+                    }
+                }
+
+                if (spanParent is UIElement parentUIElement)
+                {
```

Comment:
Rather than walking this tree for each Inline - should we just store the parent UIElement (TextBlock, RichTextBlock, or otherwise) as state for each Inline?

Reply:
@rozele Actually, the TextPointer has a member that does exactly that TextPointer.VisualParent. The reason that I use parent for traversal is that on one occasion while debugging I noticed that Parent chain gets set but VisualParent was null and only after another loop was completed it was set to the RichTextBlock control that owns the span. So I guess it is similar to the reason above that I had to post on UI Thread, the RichTextBlock implementation takes a loop to set everything after document element has been added. My take on keeping the parent as state is that since it is already available it may be good to avoid duplication. It will take memory and sometimes it may also go out-of-sync. Do you think it is better to use VisualParent here to avoid iteration? If so, I can look into it again and see if it can be used

---
In reply to: [166428345](https://github.com/Microsoft/react-native-windows/pull/1595#discussion_r166428345) [](ancestors = 166428345)